crates/polars-plan/Cargo.toml

```diff
@@ -41,8 +41,9 @@ rayon = { workspace = true }
 recursive = { workspace = true }
 regex = { workspace = true, optional = true }
 serde = { workspace = true, features = ["rc"], optional = true }
-serde_json = { workspace = true, optional = true }
+serde_json = { workspace = true, optional = false }
```

Comment:
We shouldn't add a new ArrayVec dependency. And `serde_json` should not be a required dependency here.

Adding dependencies is not something we want ideally.